### PR TITLE
Let `ijar` warn only once per unsupported attribute

### PR DESCRIPTION
**Pull request by fmeum**
Work towards #21781

### CODE REVIEW

File: third_party/ijar/test/ijar_test.sh [hunk 1]
```diff
@@ -575,6 +577,13 @@ function test_source_debug_extension_attribute() {
   expect_not_log "SourceDebugExtension" "SourceDebugExtension preserved!"
 }
 
+function test_duplicated_unsupported_attribute() {
+  # Check that unsupported attribute warnings are only emitted once
+  $IJAR $DUPLICATEDUNSUPPORTEDATTRIBUTE_JAR $DUPLICATEDUNSUPPORTEDATTRIBUTE_IJAR >& $TEST_log || fail "ijar failed"
+  expect_log 'skipping unknown attribute: "some_unknown_type"'
```
Comment (hvadehra) [inline third_party/ijar/test/ijar_test.sh:583]: Would `expect_log_once` work here?

**Review by hvadehra — COMMENTED** [review]
Thanks for this!

File: third_party/ijar/test/ijar_test.sh (hunk 1 above)
Comment (fmeum) [reply third_party/ijar/test/ijar_test.sh]: Nice, didn't know about it, changed.

**Comment by cushon** [comment]
Only warning once per attribute SGTM, thanks.

I think the idea behind the warnings is that we want to make a decision about new attributes, and either support it in ijar (if it's a standard JVM attribute that contains references to constant pool entries), or suppress the warning (if it's a proprietary attribute). For the osgi/aspectj ones in the bug, I think we could also add suppressions.

**Review by cushon — APPROVED** [review]

**Comment by fmeum** [comment]
CC @dkashyn-sfdc in case you want to submit a follow-up change to suppress these attributes.

**Review by hvadehra — APPROVED** [review]

**Comment by dkashyn-sfdc** [comment]
@fmeum will this be released as part of 7.2?

**Comment by fmeum** [comment]
If I'm not mistaken, it will be in the next rules_java release, which you can update to independently of the next Bazel release (although it will also be the default with 7.2.0).

**Comment by dkashyn-sfdc** [comment]
Let's hope this got merged till 7.2.0 GA then ;)

**Comment by dkashyn-sfdc** [comment]
@fmeum still seeing this with `7.2.0rc1`

**Comment by fmeum** [comment]
@dkashyn-sfdc Could you try to update to `rules_java` 7.6.0? The update didn't make it into rc1.

@hvadehra Are you planning to cherry-pick the update into 7.2.0?

**Comment by hvadehra** [comment]
https://github.com/bazelbuild/bazel/pull/22310 is pending internal review. After that, it'll likely get cherry-picked in rc2, if we have an rc2.

**Pull request closed without merging**